ostd/src/mm/frame/options.rs

```diff
@@ -56,13 +60,29 @@ impl FrameAllocOptions {
 
     /// Allocates a collection of page frames according to the given options.
     pub fn alloc(&self) -> Result<Vec<Frame>> {
+        let size = self.nframes * PAGE_SIZE;
         let pages = if self.is_contiguous {
-            page::allocator::alloc(self.nframes * PAGE_SIZE, |_| FrameMeta::default())
-                .ok_or(Error::NoMemory)?
-        } else {
-            page::allocator::alloc_contiguous(self.nframes * PAGE_SIZE, |_| FrameMeta::default())
+            PAGE_ALLOCATOR
+                .get()
+                .unwrap()
+                .lock()
+                .alloc(Layout::from_size_align(size, PAGE_SIZE).unwrap())
+                .map(|begin_paddr| {
+                    cont_pages::ContPages::from_unused(begin_paddr..begin_paddr + size, |_| {
+                        FrameMeta::default()
+                    })
+                })
```

Comment:
You should have implemented the `ostd::page::allocator::alloc(layout, meta_fn)` API, you should use that here.

I think it still necessary to have a set of `ostd::page::allocator::alloc_` API that wraps over the `from_unused` and the `PAGE_ALLOCATOR` call. As for the boot page allocator, accessing `BOOT_PAGE_ALLOCATOR` directly is fine.